docs/src/modules/ROOT/pages/using-timefold-solver/running-the-solver.adoc

```diff
@@ -180,76 +209,78 @@ Any variable that changed between the "before move" solution and the "after move
 Any variable that changed between the "after move" solution and "after undo move" solution without either a
 `beforeVariableChanged` or `afterVariableChanged` would be reported here.
 
-This mode is <<environmentModeReproducible,reproducible>> (see the reproducible mode).
+This mode is <<environmentModeReproducibility,reproducible>> (see the reproducible mode).
 It is also intrusive because it calls the method `calculateScore()` more frequently than a non-assert mode.
 
 The `TRACKED_FULL_ASSERT` mode is by far the slowest mode,
 because it clones solutions before and after each move.
 
 
+[#environmentModeFullAssert]
+==== `FULL_ASSERT`
+
+The `FULL_ASSERT` mode turns on all assertions (such as assert that the incremental score calculation is uncorrupted for each move) to fail-fast on a bug in a Move implementation, a constraint, the engine itself, ...
+
+This mode is <<environmentModeReproducibility,reproducible>>.
+It is also intrusive because it calls the method `calculateScore()` more frequently than a non-assert mode.
```

Comment:
```suggestion
It is also intrusive because it calls the method `calculateScore()` more frequently than a non-assert mode, making the `FULL_ASSERT` mode very slow.
```